src/CLIP.py

```diff
@@ -74,35 +85,71 @@ def forward(self,x):
         x = x + self.attn(self.ln_1(x)) # do prelayer norm, then attn, then residual connection
         x = x + self.mlp(self.ln_2(x)) # same as above, but with mlp
         return x
+
```

Comment:
The multi-head attention that is currently commented out doesn't work. I made some modifications below. I left the commented out section for visibility.